nixos/platform/network.nix

```diff
@@ -357,63 +350,115 @@ in
       # does not propagate to the network target, etc etc.
-      (listToAttrs
-        ((map (iface:
+      (listToAttrs (
+
+        (map (iface:
           (lib.nameValuePair
-            "network-link-properties-${iface.link}"
+            "${iface.link}-netdev"
             rec {
-              description = "Ensure link properties for ${iface.link}";
-              wantedBy = [ "network-addresses-${iface.interface}.service"
-                           "multi-user.target" ];
-              after = [ "sys-subsystem-net-devices-${utils.escapeSystemdPath iface.link}.device" ];
-              before = wantedBy;
-              path = [ pkgs.nettools pkgs.ethtool pkgs.procps fclib.relaxedIp ];
+              description = "Ensure network device settings for ${iface.link}";
+              wantedBy = [ "network-setup.service" "multi-user.target"  ];
+              requires = [ "network-setup.service" ];
+              path = [ pkgs.nettools pkgs.ethtool pkgs.procps fclib.relaxedIp pkgs.jq pkgs.util-linux];
               script = ''
-                LINK_DRIVER=$(ethtool -i ${iface.link} | grep "driver: " | cut -d ':' -f 2 | sed -e 's/ //')
-                case $LINK_DRIVER in
-                    e1000|e1000e|igb|ixgbe|i40e)
-                        # Set adaptive interrupt moderation. This does increase
-                        # latency.
-                        echo "Enabling adaptive interrupt moderation ..."
-                        ethtool -C "${iface.link}" rx-usecs 1 || true
-                        # Larger buffers.
-                        echo "Setting ring buffer ..."
-                        ethtool -G "${iface.link}" rx 4096 tx 4096 || true
-                        # Large receive offload to reduce small packet CPU/interrupt impact.
-                        echo "Enabling large receive offload ..."
-                        ethtool -K "${iface.link}" lro on || true
-                        ;;
-                esac
-
-                echo "Disabling flow control"
-                ethtool -A ${iface.link} autoneg off rx off tx off || true
-
-                # Ensure MTU
-                ip l set ${iface.link} mtu ${toString iface.mtu}
-
-                # Add long alternative names according to the external label
-                ip l property add altname ${quoteLabel iface.externalLabel} dev ${iface.link}
+                set -e
+
+                touch /var/lock/interface-rename.lock
+                exec 4</var/lock/interface-rename.lock
+
+                echo "Renaming to ${iface.link} ..."
+
+                echo "Acquiring interface renaming lock ..."
+                flock -x -w 60 4 || exit 1
+                echo "Got interface renaming lock ..."
+
+                # Look up the desired interface's current name by mac address
+                interface="$(ip -j -d link show | jq -r '.[] | select(.linkinfo?.info_kind? == null) | select(.address == "${iface.mac}") | .ifname')"
```

Comment:
Another thought which comes to mind: what happens if there is no interface with the desired MAC address at all? In the world of `.device` units, this would cause dependents to fail with a dependency error (after a timeout?) - presumably this script would just exit with an error somewhere.

(Handling this better could reasonably be kicked down the road to the discussed Python implementation...)

Reply:
will do, made a note in the script so i won't forget during python conversion.